### DISCUSSION TITLE
Question about rGFA segments missing optional fields

### DISCUSSION THREAD

**Discussion by cwatt** [ComparativeGenomicsToolkit/cactus#1485] [Q&A]
Hello, I'm trying to understand the GFA file format that is produced by minigraph-cactus and have a question about the optional fields SN:Z, SO:i, and SR:i. These optional fields are specific to the [rGFA format](https://github.com/lh3/gfatools/blob/master/doc/rGFA.md#the-reference-gfa-rgfa-format) which disallows overlaps.

I noticed that some segments in my GFA have these fields and others don't, like so:

```
S       23236   G       SN:Z:line#0#1    SO:i:850381     SR:i:0
S       23237   A
S       23238   CCCTAGGTGCTAGGAGGAGGCCTGCGCCGGATTTTTGCCAAGATTCGCACATTTGCGATCTCGGCGAGCTGTTGAACAAGGGCAGAAGAACACTTCCACGTGTTCGACGCGCTCTGAACGCGCCAGAGCACGCTCAGCGCCCGA        SN:Z:line#0#1    SO:i:850382   SR:i:0
```
I see that segments 23236 and 23238 are contiguous with one-another (see SO:i). Should I assume that segment 23237 is overlapping with segment 23236 (in other words, a SNP) and occupies the same coordinate, even though that information is not  recorded? This is backed-up by the fact that segment 23237 seems to take the place of 23236 in a walk:

`W  line  0       1       960733  1261738 ...>23229>23231>23232>23234>23235>23237>23238>23240>23241>...`

 If so, is it always the case that segments without stable coordinates overlap with the segment above?

Thank you!
Cassi

**Comment by glennhickey** [comment]
Yeah, cactus only adds those tags to the reference path (`line` in your case, apparently), though we start adding the tags to everything sometime soon. 

While the node id's are sorted in the graph, so `23237` is surely very near `23236`, you shouldn't draw any assumptions about what kind of variant it is (apart from it not being on the reference path).  Much better to visualize the graph, or look at the VCF output for that.
